pontoon/base/tests/test_models.py

```diff
@@ -0,0 +1,167 @@
+        inactive_contributor = UserFactory.create(email='inactive@example.com')
+
+        top_contributors = User.translators.with_translation_counts()
+        assert_true(active_contributor in top_contributors)
+        assert_true(inactive_contributor not in top_contributors)
+
+    def test_unique_translations(self):
+        """
+        Checks if contributors with identical translations are returned.
+            """
+
+        unique_translator = TranslationFactory.create().user
+        identical_translator = IdenticalTranslationFactory.create().user
+        top_contributors = User.translators.with_translation_counts()
+
+        assert_true(unique_translator in top_contributors)
+        assert_true(identical_translator not in top_contributors)
+
+
+    def test_contributors_order(self):
+        """
+        Checks if users are ordered by count of contributions.
+        """
+        users = UserFactory.create_batch(5)
+
+        for i, user in enumerate(users):
+            TranslationFactory.create_batch(5-i, user=user)
+
+        top_translations_totals = [user.translations_count for user in User.translators.with_translation_counts()]
+        assert_equal(top_translations_totals, [5, 4, 3, 2, 1])
+
+
+    def test_contributors_limit(self):
+        """
+        Checks if proper count of user is returned.
+        """
+        TranslationFactory.create_batch(110)
+
+        top_contributors = User.translators.with_translation_counts()
+
+        assert_true(len(top_contributors) == 100)
+
+    def create_contributor_with_translation_counts(self, approved, unapproved, needs_work, **kwargs):
+        """
+        Helper method, creates contributor with given translations counts.
+        """
+        contributor = UserFactory.create()
+        TranslationFactory.create_batch(approved, user=contributor, approved=True, **kwargs)
+        TranslationFactory.create_batch(unapproved, user=contributor, approved=False, fuzzy=False, **kwargs)
+        TranslationFactory.create_batch(needs_work, user=contributor, fuzzy=True, **kwargs)
+        return contributor
+
+    def test_translation_counts(self):
+        """
+        Checks if translation counts are calculated properly.
+        Tests creates 3 contributors with different numbers translations and checks if their counts match.
+        """
+
+        first_contributor = self.create_contributor_with_translation_counts(approved=7, unapproved=3, needs_work=2)
+        second_contributor = self.create_contributor_with_translation_counts(approved=5, unapproved=9, needs_work=2)
+        third_contributor = self.create_contributor_with_translation_counts(approved=1, unapproved=2, needs_work=5)
+
+        top_contributors = User.translators.with_translation_counts()
+
+        assert_equal(top_contributors.count(), 3)
+        assert_equal(top_contributors[0], second_contributor)
+        assert_equal(top_contributors[0].translations_count, 16)
```

Comment:
We have a helper that eliminates a lot of the boilerplate for asserting attributes: https://github.com/mozilla/pontoon/blob/master/pontoon/base/tests/__init__.py#L141